test/kwokctl/kwokctl_logs_test.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+DIR="$(realpath "${DIR}")"
+
+RELEASES=()
+
+function usage() {
+  echo "Usage: $0 <kube-version...>"
+  echo "  <kube-version> is the version of kubernetes to test against."
+}
+
+function args() {
+  if [[ $# -eq 0 ]]; then
+    usage
+    exit 1
+  fi
+  while [[ $# -gt 0 ]]; do
+    RELEASES+=("${1}")
+    shift
+  done
+}
+
+function test_create_cluster() {
+  local release="${1}"
+  local name="${2}"
+
+  KWOK_KUBE_VERSION="${release}" kwokctl -v=-4 create cluster --name "${name}" --timeout 10m --wait 10m --quiet-pull --config="${DIR}/logs.yaml"
+  if [[ $? -ne 0 ]]; then
+    echo "Error: Cluster ${name} creation failed"
+    exit 1
+  fi
+}
+
+function test_logs() {
+  local name="${1}"
+  local namespace="${2}"
+  local target="${3}"
+  # echo kwokctl --name "${name}" kubectl -n "${namespace}" logs "${target}" 2>&1 > /dev/null
```

Comment:
Not sure how to test this. I considered adding unit tests, but I wanted to add this to start a discussion.

I read this section: https://kubernetes.io/docs/concepts/cluster-administration/logging/#how-nodes-handle-container-logs

My best guess is to create a log file that is either in CRI logging format or in docker logging format
That is feasible in binary runtime, but it would be tricky to add logs to the container

Reply:
> Not sure how to test this. I considered adding unit tests, but I wanted to add this to start a discussion.

In my opinion, e2e can be done like this

```
echo "Multi line log" > input.log
kubectl logs --tail=1 -f podname > output.log &
echo "append a line log" >> input.log
# Then check that the last two lines of `output.log` are as expected
```

> My best guess is to create a log file that is either in CRI logging format or in docker logging format
That is feasible in binary runtime, but it would be tricky to add logs to the container

I think we need to actively mount the log files used by ClusterLogs/Logs into the container when we create the cluster using kwokctl?

Reply:
Sounds great to me! On it :relaxed: